### DISCUSSION TITLE
JFR Events

### DISCUSSION THREAD

**Discussion by Sarkie** [async-profiler/async-profiler#608] [Q&A] [answered]
Hi,

For JFR creation

profiler.sh -f profile.jfr

Is this all events that get captured, or the least?

Is this the most? 
profiler.sh -e cpu,alloc,lock -f profile.jfr

Is there a way to set all events (As I'm unsure what I care about)?

Or do I have to use list on the process and select each one?

Thank you!!

**Comment by apangin** [comment, marked as answer]
If `-e` is omitted, the default selection is `cpu`.

You can profile at most one event at a time, and optionally add `alloc` and/or `lock` when using `jfr` output.
See https://github.com/jvm-profiling-tools/async-profiler/#multiple-events for details.

**Reply by Sarkie** [reply]
Thank you! Understand now.
